File: src/main/java/mnm/mods/tabbychat/DefaultChannel.java
package mnm.mods.tabbychat;

public final class DefaultChannel extends ChatChannel {

    public static final AbstractChannel INSTANCE = new DefaultChannel();

    private DefaultChannel() {
        super("*");
    }

    @Override
    public String getDisplayName() {
        return getName();
    }

    // Don't mess with this channel
    @Override
    public void setAlias(String alias) {
    }

    @Override
    public void setPrefix(String prefix) {
    }

    @Override
    public void setPrefixHidden(boolean hidden) {
    }

    @Override
    public void setCommand(String command) {
    }
}

File: src/main/kotlin/mnm/mods/tabbychat/client/DefaultChannel.kt
package mnm.mods.tabbychat.client

import mnm.mods.tabbychat.api.Channel
import mnm.mods.tabbychat.api.ChannelType

object DefaultChannel : Channel {
    override val name = ""
    override val type = ChannelType.ROOT
    override var alias = ""

    override fun toString() = "*"
}